src/worldcereal/openeo/inference.py

```diff
@@ -120,22 +130,26 @@ def predict(self, features: np.ndarray) -> np.ndarray:
         # Prepare input data for ONNX model
         outputs = self.onnx_session.run(None, {"features": features})
 
-        # Get info on classes from the model
-        class_params = eval(
-            self.onnx_session.get_modelmeta().custom_metadata_map["class_params"]
-        )
-
         # Get classes LUT
-        LUT = dict(zip(class_params["class_names"], class_params["class_to_label"]))
+        lookup_table = self._parameters["lookup_table"]
 
         # Extract classes as INTs and probability of winning class values
         labels = np.zeros((len(outputs[0]),), dtype=np.uint16)
         probabilities = np.zeros((len(outputs[0]),), dtype=np.uint8)
         for i, (label, prob) in enumerate(zip(outputs[0], outputs[1])):
-            labels[i] = LUT[label]
+            labels[i] = lookup_table[label]
             probabilities[i] = int(prob[label] * 100)
 
-        return np.stack([labels, probabilities], axis=0)
+        # Extract per class probabilities
+        output_probabilities = []
+        for output_px in outputs[1]:
+            output_probabilities.append(list(output_px.values()))
+
+        output_probabilities = (np.array(output_probabilities) * 100).astype(np.uint8)
+
+        return np.hstack(
```

Comment:
if no postprocessing is requested, we'll still get all the probabilities back? Shouldn't we then just get the default [classification, probability]?

Reply:
@GriffinBabe  what about this one?